File: AnuitexTraining.BusinessLogicLayer/Models/PrintingEditions/PrintingEditionFilterModel.cs
﻿using System;
using System.Collections.Generic;
using static AnuitexTraining.Shared.Enums.Enums;

namespace AnuitexTraining.BusinessLogicLayer.Models.PrintingEditions
{
    public class PrintingEditionFilterModel
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public double MinPrice { get; set; }
        public double MaxPrice { get; set; }
        public List<PrintingEditionType> Types { get; set; }
        public CurrencyType Currency { get; set; }
        public DateTime CreationDate { get; set; }
        public string SearchString { get; set; }
    }
}

File: AnuitexTraining/ClientApp/src/app/modules/administrator/models/printing-edition-filter.model.ts
import {PrintingEditionTypeEnum} from "../../shared/enums/printing-edition-type.enum";
import {CurrencyTypeEnum} from "../../shared/enums/currency-type.enum";

export interface PrintingEditionFilterModel {
  id: number,
  title: string,
  description: string,
  types: PrintingEditionTypeEnum[],
  authors: string[],
  minPrice: number,
  maxPrice: number,
  currency: CurrencyTypeEnum,
  searchString: string
}
